### PR TITLE
feat: Keyblock microblocks table

### LINKED ISSUE #369: Keyblock microblocks table
In the keyblock details page here should also be a table with the microblocks of the keyblock with the following columns:

- Microblock hash
- Time
- Transactions count

### PR DESCRIPTION
**Pull request by janmichek**
## Description
resolves #369

## Demo

https://github.com/aeternity/aescan/assets/15363559/c48725cb-3587-4aac-bf95-5c83289178bc

## Checklist:

- [x] I have read and followed the [Contributing Guide](../../CONTRIBUTING.md)  
- [x] list block's mircroblocks

### CODE REVIEW

File: src/components/KeyblockMicroblocksTable.vue [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        <hint-tooltip>
+          {{ keyblocksHints.transactionsCount }}
+        </hint-tooltip>
+      </th>
+    </tr>
+    <tr
+      v-for="microblock in microblocks.data"
+      :key="microblock.hash">
+      <td>
+        <app-link :to="`/micro-blocks/${microblock.hash}`">
```
Comment (michele-franchi) [inline src/components/KeyblockMicroblocksTable.vue:27]:
```suggestion
        <app-link :to="`/microblocks/${microblock.hash}`">
```

The link is incorrect

**Review by michele-franchi — CHANGES_REQUESTED** [review]
Good job, I have some comments.

Any reason why the hash is not truncated? It would be the only table like this.
image

It's not clear that in general these are microblocks, it is not written anywhere.

**Comment by janmichek** [comment]
@michele-franchi 

>Any reason why the hash is not truncated? It would be the only table like this.

I was following the visual material. No strong preference here


>It's not clear that in general these are microblocks, it is not written anywhere.

I was also following the visual material. Here I would prefer label as well.

What do you think about those changes @danilosierrac ?

File: src/components/KeyblockMicroblocksTable.vue (hunk 1 above)
Comment (janmichek) [reply src/components/KeyblockMicroblocksTable.vue]: fixed

File: src/components/KeyblockMicroblocksPanel.vue [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import PaginatedContent from '@/components/PaginatedContent'
+import KeyblockMicroblocksTable from '@/components/KeyblockMicroblocksTable'
+import KeyblockMicroblocksTableCondensed from '@/components/KeyblockMicroblocksTableCondensed'
+import { isDesktop } from '@/utils/screen'
+
+const { keyblockMicroblocks: microblocks, keyblockDetails } = storeToRefs(useKeyblockDetailsStore())
+const { fetchKeyblockMicroblocks } = useKeyblockDetailsStore()
+const route = useRoute()
+const limit = computed(() => process.client && isDesktop() ? 10 : 3)
+fetchKeyblockMicroblocks({
```
Comment (lukeromanowicz) [inline src/components/KeyblockMicroblocksPanel.vue:35]: I think this should be fetched on the client side only. Right now it's fetched also on server-side but the SSR doesn't wait for the response so you don't see any warnings

**Review by lukeromanowicz — APPROVED** [review]
Looks really good. Thank you for implementing it! I left one small note but I approve in advance

**Review by michele-franchi — CHANGES_REQUESTED** [review]
Please have a look at the recent changes I suggested in the design:
- A new tab is visible called "Microblocks"
I would still use truncated hashes for consistency but I don't have a strong preference.

File: src/components/KeyblockMicroblocksPanel.vue (hunk 2 above)
Comment (janmichek) [reply src/components/KeyblockMicroblocksPanel.vue]: Thanks, fixed

**Comment by janmichek** [comment]
> * A new tab is visible called "Microblocks"

Ok, both adjusted

**Review by michele-franchi — APPROVED** [review]
Good job, thanks for the fixes 👏

**Review by CharisSiarampalis — APPROVED** [review]

**Pull request merged**